java-api-core/src/com/kinvey/java/core/MediaHttpDownloader.java

```diff
@@ -270,7 +270,7 @@ public FileMetaData download(FileMetaData metaData, OutputStream out) throws IOE
             }
 
             // set Range header (if necessary)
-            if (bytesDownloaded != 0 || currentRequestLastBytePos != -1) {
+            if (metaData.getSize() != 0 && (bytesDownloaded != 0 || currentRequestLastBytePos != -1)) {
```

Comment:
Per conversation with Yuliya:
"if the file is empty, we’re getting bytesDownloaded = 0, but currentRequestLastBytePos != -1 because chunkSize != 0, that’s why it gets into the if. But we can’t simplify it to: if (bytesDownloaded != 0 && currentRequestLastBytePos != -1) , for example, because it won’t be working for non-empty files"